### PR TITLE
LPS-158346 [Raylife AP] Creation of Accident Citation box

### PR DESCRIPTION
**Pull request by eliasantosf**
Sub-task: [LPS-158346](https://issues.liferay.com/browse/LPS-158346)

### CODE REVIEW

**Comment by eliasantosf** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/context/NewApplicationAutoContextProvider.tsx [hunk 1]
```diff
@@ -368,7 +385,65 @@ const reducer = (state: InitialStateTypes, action: ApplicationActions) => {
...
-						...state.steps.vehicleInfo,
+						...state.steps.driverInfo,
+						form: payload,
+					},
+				},
+			};
+		}
+
+		case ACTIONS.SET_NEW_ACCIDENT_CITATION: {
+			const payload = state.steps.driverInfo.form;
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/context/NewApplicationAutoContextProvider.tsx:396]:
```suggestion
			const payload = state?.steps?.driverInfo?.form;
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/context/NewApplicationAutoContextProvider.tsx [hunk 2]
```diff
@@ -368,7 +385,65 @@ const reducer = (state: InitialStateTypes, action: ApplicationActions) => {
...
+
+			const forms = payload.filter(
+				(form) => form.id === action.payload.id
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/context/NewApplicationAutoContextProvider.tsx:399]:
```suggestion
				(form) => form.id === action?.payload?.id
```

**Review by joanagrego — CHANGES_REQUESTED** [review]
Great job! I added some suggestions :)

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/context/NewApplicationAutoContextProvider.tsx (hunk 2 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/context/NewApplicationAutoContextProvider.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/context/NewApplicationAutoContextProvider.tsx (hunk 1 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/context/NewApplicationAutoContextProvider.tsx]: Done!

**Comment by eliasantosf** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx [hunk 3]
```diff
@@ -610,12 +697,81 @@ const DriverInfoForm = ({form, formNumber, id}: FormDriverInfoTypes) => {
...
+							label="No"
+							onClick={() => setAccidentCitationValue(false)}
+							value="no"
+						/>
 					</ClayRadioGroup>
 				</div>
+
+				{accidentCitationValue && (
+					<div className="row">
+						{payload[formIndex].map(
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx:717]: payload needs more context.

**Review by silasbsi — CHANGES_REQUESTED** [review]
LGTM :1st_place_medal: But I added some suggestions

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/context/NewApplicationAutoContextProvider.tsx [hunk 4]
```diff
@@ -368,7 +385,64 @@ const reducer = (state: InitialStateTypes, action: ApplicationActions) => {
...
+			};
+		}
+
+		case ACTIONS.SET_NEW_ACCIDENT_CITATION: {
+			const payload = state?.steps?.driverInfo?.form;
+			const formId = action?.payload?.id;
+
+			const forms = payload.filter((form) => form.id === formId);
+
+			forms.map((form) =>
```
[the remark this answers is not in the document]
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/context/NewApplicationAutoContextProvider.tsx]: Done!

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx (hunk 3 above)
Comment (eliasantosf) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/DriverInfo/DriverInfoForm.tsx]: Done!

**Comment by eliasantosf** [comment]
ci:test:sf

**Review by silasbsi — APPROVED** [review]

**Review by joanagrego — APPROVED** [review]
Great job! :up:

**Comment by silasbsi** [comment]
ci:forward

**Pull request closed without merging**